### PR TITLE
simplify boolean expressions

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: src/liballoc/collections/binary_heap.rs [hunk 1]
```diff
@@ -536,7 +536,7 @@ impl<T: Ord> BinaryHeap<T> {
             while child < end {
                 let right = child + 1;
                 // compare with the greater of the two children
-                if right < end && !(hole.get(child) > hole.get(right)) {
+                if right < end && (hole.get(child) <= hole.get(right)) {
```
Comment (Mark-Simulacrum) [inline src/liballoc/collections/binary_heap.rs]: Hm, extra parentheses?

**Review by Mark-Simulacrum — APPROVED** [review]
r=me with parentheses fixed

File: src/liballoc/collections/binary_heap.rs (hunk 1 above)
Comment (matthiaskrgr) [reply src/liballoc/collections/binary_heap.rs]: Fixed, thanks!

File: src/liballoc/collections/binary_heap.rs [hunk 2]
```diff
@@ -568,7 +568,7 @@ impl<T: Ord> BinaryHeap<T> {
...
```
Comment (JohnTitor) [inline src/liballoc/collections/binary_heap.rs]: Same here, unnecessary parentheses.

Comment (matthiaskrgr) [reply src/liballoc/collections/binary_heap.rs]: I guess the `unused-parens` lint only warns on outermost parenthesis?

**Pull request merged**